### PR TITLE
Add on device summary sample

### PR DESCRIPTION
**Pull request by sebastianbenz**

### CODE REVIEW

**Review by oliverdunk — CHANGES_REQUESTED** [review]
Got this running, it's really great! Nice work :)

File: functional-samples/ai.gemini-on-device-summarization/manifest.json [hunk 1]
```diff
@@ -0,0 +1,23 @@
+{
+  "name": "Summarizer",
+  "version": "0.1",
+  "manifest_version": 3,
+  "description": "Get a short summary of every webpage.",
+  "background": {
+    "service_worker": "background.js"
+  },
+  "permissions": ["tabs", "scripting", "sidePanel", "storage"],
+  "host_permissions": ["http://*/*", "https://*/*"],
```
[the remark this answers is not in the document]
Comment (sebastianbenz) [reply functional-samples/ai.gemini-on-device-summarization/manifest.json:10]: Done. You're right, I'm doing the protocol check in the sw anyway.

File: functional-samples/ai.gemini-on-device-summarization/sidepanel/index.css [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+.warning {
+  background: #ffc6c6;
+  padding: 16px;
+  border-radius: 8px;
+  margin: 16px 0;
+}
+
+/* CSS via https://projects.verou.me/bubbly/ */
+.speech-bubble {
```
[the remark this answers is not in the document]
Comment (sebastianbenz) [reply functional-samples/ai.gemini-on-device-summarization/sidepanel/index.css]: Removed the bubble part, didn't even make that much sense anyway...

File: functional-samples/ai.gemini-on-device-summarization/README.md [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+## Overview
+
+The extension summarizes the content of the currently open tab. It uses Mozilla's [readability](https://github.com/mozilla/readability) library to extract the content of the currently active tab and displays a summary of the page generated by [Chrome's experimental summarization API](https://developer.chrome.com/blog/august2024-summarization-ai) in a side panel.
+
+## Running this extension
+
+1. Clone this repository
+2. Run `npm install` in this folder to install all dependencies.
+3. Run `npm run build` to bundle the content script .
+4. Load this directory in Chrome as an [unpacked extension](https://developer.chrome.com/docs/extensions/mv3/getstarted/development-basics/#load-unpacked).
```
[the remark this answers is not in the document]
Comment (sebastianbenz) [reply functional-samples/ai.gemini-on-device-summarization/README.md]: done

**Review by sebastianbenz — COMMENTED** [review]
PTAL

File: functional-samples/ai.gemini-on-device-summarization/background.js [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+});
+
+async function showSummary(tabId) {
+  const tab = await chrome.tabs.get(tabId);
+  if (!tab.url.startsWith('http')) {
+    return;
+  }
+  const injection = await chrome.scripting.executeScript({
+    target: { tabId },
+    files: ['dist/scripts/extract-content.js']
```
Comment (oliverdunk) [inline functional-samples/ai.gemini-on-device-summarization/background.js:19]: Does this need updating?

**Review by oliverdunk — CHANGES_REQUESTED** [review]

File: functional-samples/ai.gemini-on-device-summarization/background.js (hunk 4 above)
Comment (sebastianbenz) [reply functional-samples/ai.gemini-on-device-summarization/background.js]: yes!

**Review by oliverdunk — APPROVED** [review]
LGTM, thanks!

**Pull request merged**